packages/notifi-graphql/lib/NotifiSubscriptionService.ts

```diff
@@ -0,0 +1,82 @@
+import { createClient, Client } from 'graphql-ws';
+export class NotifiSubscriptionService {
+  public wsClient: Client | undefined;
+  constructor(private wsurl: string) {
+  }
+
+  disposeClient = () => {
+    if (this.wsClient) {
+      this.wsClient.terminate();
+      this.wsClient.dispose();
+    }
+  };
+
+  subscribe = (jwt: string | undefined, subscriptionQuery: string, actionOnMessageReceived: () => void | undefined) => {
+    this.initializeClientIfUndefined(jwt);
+    this.wsClient?.subscribe({
+      query: subscriptionQuery,
+      extensions: {
+        type: 'start'
+      }
+    },
+      {
+        next: (data) => {
+          if (actionOnMessageReceived) {
+            actionOnMessageReceived();
+          }
+          console.log("Subscription data recieved" + JSON.stringify(data));
+        },
+        error: (error) => {
+          console.error('Subscription error:', error);
+        },
+        complete: () => {
+          console.log('Subscription completed');
+        },
+      })
+  }
+
+  private initializeClientIfUndefined = (jwt: string | undefined) => {
+    if (!this.wsClient) {
+      this.initializeClient(jwt);
+    }
+  }
+
+  private initializeClient = (jwt: string | undefined) => {
+    this.wsClient = createClient({
+      url: this.wsurl,
+      connectionParams: {
+        Authorization: `Bearer ${jwt}`,
+      },
+    });
+
+    this.wsClient.on('connected', () => {
+      console.log('WebSocket connection opened');
+    });
+
+    this.wsClient.on('error', (error) => {
+      console.error('WebSocket error:', error);
+    });
+
+    this.wsClient.on('closed', (event) => {
+      setTimeout(() => {
+        // this.ensureWebsocketProviderSetup();
+      }, 3000);
+    });
+
+    this.wsClient.on('message', (data) => {
+      if (data.type == 'next') {
+        console.log('Notification history has been changed: ', data);
+      } else {
+        console.log('Received data:', data);
+      }
+    });
```

Comment:
Similarly, shall we also exposes these:
```ts
onConnected?: () => void
onConnectionError?: (error)=> void
onConnectionClosed?: (event)=> void
onMessage: (data) => void // Not sure if it has any difference from the `.subscribe`' `next` argument
```

Reply:
We are already handling at subscribe query level so do we really need to expose this also, it is kind of redundant and also initializing of client we are doing inside this class and also Wsclient is private.